src/main/java/io/candydoc/ddd/tactical_pattern/DomainRepository.java

```diff
@@ -0,0 +1,24 @@
+package io.candydoc.ddd.tactical_pattern;
+
+import io.candydoc.ddd.model.DDDPattern;
+import java.util.Set;
+import lombok.EqualsAndHashCode;
+import lombok.ToString;
+import lombok.experimental.SuperBuilder;
+
+@SuperBuilder
+@EqualsAndHashCode(callSuper = true)
+@ToString(callSuper = true)
+public class DomainRepository extends TacticalModel {
+  private static final Set<DDDPattern> ALLOWED_DEPENDENCIES = Set.of(TacticalPattern.AGGREGATE);
```

Comment:
It can also depend on domain entities

Reply:
I think that repository may more rely on a value object as an entity ID. That's why we may consider allowing value object dependencies rather than entity dependencies

Reply:
Nope. A repository allow the access of an _identified_ domain model. In the tactical patterns, only Aggregates and Entities follow this rules (an Aggregate is a kind of entity btw).